src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -1185,6 +1225,26 @@ def forward(
         ```"""
         return_dict = return_dict if return_dict is not None else self.config.use_return_dict
 
+
+        # create causal mask
+        # [bsz, seq_len] -> [bsz, 1, tgt_seq_len, src_seq_len]
+
+        if self.config.condition_on_previous_text or torch.all(decoder_input_ids == self.config.decoder_start_token_id):
+            self.condition_start_idx = torch.zeros_like(decoder_input_ids).int()
+            self.last_is_timestamp = None
+
+        if not self.config.condition_on_previous_text:
+            timestamp_tokens: torch.Tensor = decoder_input_ids.ge(self.config.ts_start_token_id)
```

Comment:
Very nicely done! Only a small nit here (we don't include typehints in-line)

```suggestion
            timestamp_tokens = decoder_input_ids.ge(self.config.ts_start_token_id)
```